### ISSUE TITLE
Random BDD test failure

### ISSUE THREAD

**Issue by benjaoming** [learningequality/ka-lite#4716]
Having opened a lot of PRs (25 in a day), I found that I had to run the same tests several times because of this BDD test failure:

```
  Scenario: Content is available        # kalite/distributed/features/learn_content.feature:9
    Given I open some available content # kalite/distributed/features/steps/learn_content.py:17
    Given I open some available content # kalite/distributed/features/steps/learn_content.py:17 4.449s
    Then I should see no alert          # kalite/distributed/features/steps/learn_content.py:33
    Then I should see no alert          # kalite/distributed/features/steps/learn_content.py:33 1.732s
      Assertion Failed: NoSuchElementException not raised
```

It occurs alone. Often after dark. It's very hard to spot because it's very camouflaged. Only a vague red stripe distinguishes it.

IMO we should get it fixed before releasing because it's going to be annoying for both this release and any future fixes.

@aronasorman please re-assign at will :)

**Comment by MCGallaspy** [comment]
I noticed this too.

**Comment by aronasorman** [comment]
I got a similar behaviour in the 0.15.1 PR.

**Comment by MCGallaspy** [comment]
Here's the relevant steps as they appear on `develop` now:

```python
@given("I open some available content")
def step_impl(context):
    update_item(path="khan/math/arithmetic/addition-subtraction/basic_addition/addition_1/", update={"available": True})
    context.browser.get(build_url(context, reverse("learn") + "khan/math/arithmetic/addition-subtraction/basic_addition/addition_1/"))

@then(u'I should see no alert')
def impl(context):
    assert_no_element_by_css_selector(context, ".alert")
```

My only thought is that there is a race condition with `update_item` -- that the requested page does not reflect the updated db row. Not sure how to test this hypothesis. Any insight here @rtibbles?

**Comment by MCGallaspy** [comment]
Strange, here's the screencast when it passes: https://saucelabs.com/jobs/b8d864eed3af4bea9daff749f63d81ae

Note the alert indicating the content was not found.

**Comment by benjaoming** [comment]
```


  @as_learner
  Scenario: User alters a star rating       # kalite/distributed/features/content_rating.feature:12
    Given I have filled out a feedback form # kalite/distributed/features/steps/content_rating.py:115
    Given I have filled out a feedback form # kalite/distributed/features/steps/content_rating.py:115 7.714s
    When I alter a star rating              # kalite/distributed/features/steps/content_rating.py:43
    When I alter a star rating              # kalite/distributed/features/steps/content_rating.py:43 2.498s
    Then the altered rating is displayed    # kalite/distributed/features/steps/content_rating.py:52
    Then the altered rating is displayed    # kalite/distributed/features/steps/content_rating.py:52 0.843s
      Assertion Failed: Star rating has value 3, which should be different from the old value 3

Link to your job: https://saucelabs.com/jobs/59a8ec0681ee4a898995b6c890e13ecf
```

**Comment by rtibbles** [comment]
Fixed in #4746
